### PR TITLE
seperates task and container override

### PR DESCRIPTION
**Pull request by jeanluciano**
Sepererates the way we override task level cpu and memory from the prefect container. If no container cpu and/or memory is passed, it uses task level.

### Checklist

- [ x ] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-aws/issues/new/choose) first.
- [ x ] Includes tests or only affects documentation.
- [x ] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.

### CODE REVIEW

File: tests/workers/test_ecs_worker.py [hunk 1]
```diff
@@ -522,15 +522,23 @@ async def test_launch_types(
 @pytest.mark.parametrize(
     "cpu,memory", [(None, None), (1024, None), (None, 2048), (2048, 4096)]
 )
+@pytest.mark.parametrize("container_cpu,container_memory", [(1024, 2048)])
```
Comment (kevingrismore) [inline tests/workers/test_ecs_worker.py:525]: If we expect that `cpu` and `memory` will be inherited from the task to the container, we should add cases to the test matrix where the container cpu and memory are empty and verify that

File: prefect_aws/workers/ecs_worker.py [hunk 2]
```diff
@@ -450,6 +452,25 @@ class ECSVariables(BaseVariables):
             f"{ECS_DEFAULT_MEMORY} will be used unless present on the task definition."
         ),
     )
+    container_cpu: int = Field(
+        title="CPU",
+        default=None,
+        description=(
+            "The amount of CPU to provide to the Prefect container. Valid amounts are "
+            "specified in the AWS documentation. If not provided, a default value of "
+            f"{ECS_DEFAULT_CPU} will be used unless present on the task definition."
```
Comment (kevingrismore) [inline prefect_aws/workers/ecs_worker.py:461]: I think it would be good to make the inheritance path clearer here.

Something like this:
```
            "The amount of CPU to provide to the Prefect container. Valid amounts are "
            "specified in the AWS documentation. If not provided, the value supplied to `cpu` "
            f"for the task will be used. If neither are provided, {ECS_DEFAULT_CPU} will be used "
            "unless present on the task definition."
```

File: prefect_aws/workers/ecs_worker.py [hunk 3]
```diff
@@ -450,6 +452,25 @@ class ECSVariables(BaseVariables):
...
+        ),
+    )
+    container_memory: int = Field(
+        default=None,
+        description=(
+            "The amount of memory to provide to the Prefect container. Valid amounts"
+            " are specified in the AWS documentation. If not provided, a default value"
+            f" of {ECS_DEFAULT_MEMORY} will be used unless present on the task"
+            " definition."
```
Comment (kevingrismore) [inline prefect_aws/workers/ecs_worker.py:470]: Same as my comment for CPU, but for memory.

**Review by kevingrismore — CHANGES_REQUESTED** [review]

**Comment by kevingrismore** [comment]
I think it'll be important for us to keep in mind that this'll make it easier for users to accidentally supply values to a single container that are larger than the values supplied to the task, especially when they leave the task-level resource settings empty and then request resources for their container that are larger than the default.

I'm ok with letting AWS raise the error for us in cases where users specify many of their own containers and get the math wrong, but the case above seems like an easy pitfall for those not familiar with ECS.

@jeanluciano Do you think we should check if the Prefect container's resources are larger than the task's resources and reject the run with a nice error? Or maybe add a note on the new settings that they can't exceed the task level resources.
